lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -180,22 +192,60 @@ mReadStakeDistribution epoch db@PoolDatabase{distributions} =
     , db
     )
 
-mPutPoolRegistration :: SlotId -> PoolRegistrationCertificate -> ModelPoolOp ()
-mPutPoolRegistration sl registration db@PoolDatabase{owners,registrations} =
+mPutPoolRegistration
+    :: CertificatePublicationTime
+    -> PoolRegistrationCertificate
+    -> ModelPoolOp ()
+mPutPoolRegistration sp registration db =
     ( Right ()
     , db { owners = Map.insert poolId poolOwners owners
-         , registrations = Map.insert (sl, poolId) registration registrations
+         , registrations = Map.insert (sp, poolId) registration registrations
          }
     )
   where
-    PoolRegistrationCertificate { poolId , poolOwners } = registration
+    PoolDatabase {owners, registrations} = db
+    PoolRegistrationCertificate {poolId, poolOwners} = registration
+
+mReadPoolRegistration
+    :: PoolId
+    -> ModelPoolOp
+        (Maybe (CertificatePublicationTime, PoolRegistrationCertificate))
+mReadPoolRegistration poolId db =
+    ( Right
+        $ fmap (first fst)
+        $ Map.lookupMax
+        $ Map.filterWithKey (only poolId) registrations
+    , db
+    )
+  where
+    PoolDatabase {registrations} = db
+    only k (_, k') _ = k == k'
 
-mReadPoolRegistration :: PoolId -> ModelPoolOp (Maybe PoolRegistrationCertificate)
-mReadPoolRegistration poolId db@PoolDatabase{registrations} =
-    ( Right $ fmap snd $ Map.lookupMax $ Map.filterWithKey (only poolId) registrations
+mPutPoolRetirement
+    :: CertificatePublicationTime
+    -> PoolRetirementCertificate
+    -> ModelPoolOp ()
+mPutPoolRetirement sp retirement db =
```

Comment:
Given that the type is CertificatePublicationTime, we should probably also rename sp to cpt for consistency.

```suggestion
mPutPoolRetirement cpt retirement db =
```

Reply:
Fixed in a75c879aabde68edcc1389a38a9e09b0bbdf17a7.